lib/elixir/src/elixir_lexical.erl

```diff
@@ -1,27 +1,29 @@
 %% Module responsible for tracking lexical information.
 -module(elixir_lexical).
--export([run/2, with_file/3, trace/2, format_error/1]).
+-export([run/3, with_file/3, trace/2, format_error/1]).
 -include("elixir.hrl").
 
 -define(tracker, 'Elixir.Kernel.LexicalTracker').
 
-run(#{tracers := Tracers} = E, Callback) ->
+run(#{tracers := Tracers} = E, ExecutionCallback, AfterExecutionCallback) ->
   case elixir_config:get(bootstrap) of
     false ->
       {ok, Pid} = ?tracker:start_link(),
+      LexEnv = E#{lexical_tracker := Pid, tracers := [?MODULE | Tracers]},
 
-      try Callback(E#{lexical_tracker := Pid, tracers := [?MODULE | Tracers]}) of
+      try ExecutionCallback(LexEnv) of
         Res ->
           warn_unused_aliases(Pid, E),
           warn_unused_imports(Pid, E),
+          AfterExecutionCallback(LexEnv),
           Res
       after
         unlink(Pid),
         ?tracker:stop(Pid)
       end;
 
     true ->
-      Callback(E)
+      ExecutionCallback(E)
```

Comment:
Shouldn't the AfterExecutionCallback be invoked here too?

Reply:
Good catch, I missed that!